### ISSUE TITLE
Add src to .npmignore

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#78]
### Description of Issue
A common pattern for npm modules is to add the lib(es5) directory to the .gitignore and add the src(es6) directory to the .npmignore. [Babel is an example of this pattern](https://github.com/babel/babel/tree/7.0/packages/babel-core) that also uses lerna. 

More information can be found in this article:
http://jamesknelson.com/writing-npm-packages-with-es6-using-the-babel-6-cli/

- [x] Add a .npmignore file to each package that contains the src directory.

**Comment by Matt-Butler** [comment]
Blocked on https://github.com/cerner/terra-ui/issues/98

**Comment by bjankord** [comment]
This is partially complete now with #295 resolved. We've talked about not adding `lib` to `.gitignore` as we use files in `lib` with `terra-site` to do some testing. I think its possible we could add `lib` to `.gitignore`, we'd just want to generate those files whenever we run tests that reference them.

I'm thinking we close this and open a new issue for discussing what we want to do with `lib`. @Matt-Butler what are your thoughts?

**Comment by Matt-Butler** [comment]
+1 to adding a new issue to add lib to the .gitignore
